### PR TITLE
cli: add async execution between separate paths

### PR DESCRIPTION
**Pull request by GeoSot**

### CODE REVIEW

File: cli.js [hunk 1]
```diff
@@ -14,14 +14,30 @@ commander
...
 
-    let unusedList = [];
+    const unusedList = [];
 
-    args.forEach(arg => {
+    const executions = args.map(path => execute(path, ignore));
+
+    const results = Promise.all(executions);
+    await results.then(results => results.forEach(result => {
+        unusedList.concat(result.value);
```
Comment (XhmikosR) [inline cli.js:28]:
```suggestion
        unusedList = [...unusedList, result.value];
```

This should be better?

**Comment by XhmikosR** [comment]
@GeoSot wouldn't it better to use async/await instead of Promises? You should also catch any errors BTW:

```
C:\Users\xmr\Desktop\find-unused-sass-variables>node cli.js ../bootstrap/i-dont-exist
Looking for unused variables
Finding unused variables in "C:\Users\xmr\Desktop\bootstrap\i-dont-exist"...
(node:4784) UnhandledPromiseRejectionWarning: Error: "C:\Users\xmr\Desktop\bootstrap\i-dont-exist": Not a valid directory!
    at Object.findUnusedVars [as find] (C:\Users\xmr\Desktop\find-unused-sass-variables\index.js:25:15)
    at C:\Users\xmr\Desktop\find-unused-sass-variables\cli.js:46:33
    at new Promise (<anonymous>)
    at execute (C:\Users\xmr\Desktop\find-unused-sass-variables\cli.js:40:12)
    at C:\Users\xmr\Desktop\find-unused-sass-variables\cli.js:24:41
    at Array.map (<anonymous>)
    at main (C:\Users\xmr\Desktop\find-unused-sass-variables\cli.js:24:29)
    at Object.<anonymous> (C:\Users\xmr\Desktop\find-unused-sass-variables\cli.js:61:5)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:4784) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
(node:4784) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

Do you plan to do the same for the API? Not sure if we can provide both sync and async API, so if not, it will be a breaking change. I don't mind, just pointing it out.

**Comment by GeoSot** [comment]
I don't know any way to use async with multiple executions. Only `Promise.all` can handle them

**Comment by XhmikosR** [comment]
> Only `Promise.all` can handle them

Yeah, Promise.all should be what we need here, have a look at https://github.com/twbs/icons/blob/main/build/build-svgs.js for example.

BTW please try to keep the changes to the minimum ones so that we don't have any unrelated changes :)

**Comment by GeoSot** [comment]
I am trying. Please be my guest to change anything

File: cli.js [hunk 2]
```diff
@@ -14,14 +14,41 @@ commander
...
+}
+
+const showResults = (unusedCount = 0) => {
+    if (unusedCount > 0) {
+        console.log(chalk.redBright(`${unusedCount} unused variables found`));
+        process.exit(1);
+    }
+
+    console.log(chalk.greenBright('No unused variables found!'));
+    process.exit(0);
```
Comment (XhmikosR) [inline cli.js:47]: I wonder if we really need this? IIRC error code should be zero if everything went fine, right? We should remove it later if it's not needed :)

**Review by XhmikosR — APPROVED** [review]

**Pull request merged**